clicommand/agent_start.go

```diff
@@ -511,9 +514,20 @@ var AgentStartCommand = cli.Command{
 		},
 		cli.BoolFlag{
 			Name:   "no-command-eval",
-			Usage:  "Don't allow this agent to run arbitrary console commands, including plugins",
+			Usage:  "Don't allow this agent to run arbitrary console commands; also disables plugins by default - can only be used with command-mode 'legacy'",
 			EnvVar: "BUILDKITE_NO_COMMAND_EVAL",
 		},
+		cli.StringFlag{
+			Name:   "command-mode",
+			Value:  job.CommandModeLegacy,
+			Usage:  "Specifies how the agent should run commands - can be 'legacy', 'shell' or 'program'",
+			EnvVar: "BUILDKITE_COMMAND_MODE",
+		},
+		cli.BoolTFlag{
+			Name:   "command-repo-only",
+			Usage:  "Command must refer to a program in the repo - only used if 'command-mode' is 'program'",
+			EnvVar: "BUILDKITE_COMMAND_REPO_ONLY",
+		},
```

Comment:
> if other options come into existence

Seems like a big "if", and even then it might not make sense to use them together. But I agree, `program-repo-only-new-option` is so bad that we should avoid it. So, keeping it as an option to program mode looks good to me.